common/src/main/java/io/github/axolotlclient/util/CachedAPI.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2024 moehreag <moehreag@gmail.com> & Contributors
+ *
+ * This file is part of AxolotlClient.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * For more information, see the LICENSE file.
+ */
+package io.github.axolotlclient.util;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import org.jetbrains.annotations.ApiStatus;
+
+@SuppressWarnings("UnstableApiUsage")
+public final class CachedAPI<K, V> {
+    public interface APIHandler<K, V> {
+        CompletableFuture<V> makeRequest(K key);
+    }
+
+    private final LoadingCache<K, CompletableFuture<V>> cache = CacheBuilder
+        .newBuilder()
+        .maximumSize(100)
```

Comment:
It might make sense to allow different instances of `CachedAPI` to have different maximum sizes since f.e. the level data will usually reach multiple hundred players quite quickly